Control/public/Model.js

```diff
@@ -94,22 +93,22 @@ export default class Model extends Observable {
    * @returns {object} User's role 
    */ 
   setRole() {
-    let role = this.Roles.Guest;
+    let role = ROLES.Guest;
```

Comment:
I think this is a `get` rather than a `set` method. Also there is no need for variable `role` to be defined. 
Is it only us adding `det-` prefix? If not maybe it would be good to make the role and prefix uppercase before comparison.

It can be written with return statements:
```
getRole() {
		if (this.session.access.includes('admin')) {
      return ROLES.Admin;
    } else if (this.session.access.includes('global')) {
      return ROLES.Global;
    } else if (this.session.access.some((role) => role.toUpperCase().startsWith(PREFIX.SSO_DET_ROLE.toUpperCase()))) {
      return ROLES.Detector;
    }
    return ROLES.Guest;
  }
```